src/components/conditional.js

```diff
@@ -26,8 +26,13 @@
     const logic = useLogic(displayLogic);
 
     const evalCondition = () => {
-      const leftAsNumber = parseFloat(leftValue);
-      const rightAsNumber = parseFloat(rightValue);
+      const [leftParsed, rightParsed] =
+        (typeof leftValue === 'string' || typeof leftValue === 'number') &&
+        (typeof rightValue === 'string' || typeof rightValue === 'number') &&
+        !isNaN(leftValue) &&
+        !isNaN(rightValue)
+          ? [parseFloat(leftValue), parseFloat(rightValue)]
+          : [leftValue.toString(), rightValue.toString()];
```

Comment:
Is it an idea to split this up into separate functions? There's a lot going on here